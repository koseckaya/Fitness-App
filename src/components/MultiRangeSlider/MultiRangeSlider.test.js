import React from 'react';
import renderer from 'react-test-renderer';
import MultiRangeSlider from './MultiRangeSlider';

const props = {
    min: 0, 
    max: 200, 
    onChange: () => {},
}

describe("MultiRangeSlider component", () => {
    it('should render component with default props', () => {
        const tree = renderer
            .create(<MultiRangeSlider {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render component without props', () => {
        const newProps = {...props, min: 200, max: 0, }
        const component = renderer
            .create(<MultiRangeSlider {...newProps} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
})