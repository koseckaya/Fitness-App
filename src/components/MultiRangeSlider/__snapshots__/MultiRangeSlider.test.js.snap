// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiRangeSlider component should render component with default props 1`] = `
<div
  className="slider-container"
>
  <input
    className="thumb thumb--left"
    max={200}
    min={0}
    onChange={[Function]}
    style={
      Object {
        "zIndex": 0,
      }
    }
    type="range"
    value={0}
  />
  <input
    className="thumb thumb--right"
    max={200}
    min={0}
    onChange={[Function]}
    type="range"
    value={200}
  />
  <div
    className="slider"
  >
    <div
      className="slider__track"
    />
    <div
      className="slider__range"
    />
    <div
      className="slider__left-value"
    >
      0
    </div>
    <div
      className="slider__right-value"
    >
      200
    </div>
  </div>
</div>
`;

exports[`MultiRangeSlider component should render component without props 1`] = `
<div
  className="slider-container"
>
  <input
    className="thumb thumb--left"
    max={0}
    min={200}
    onChange={[Function]}
    style={
      Object {
        "zIndex": 5,
      }
    }
    type="range"
    value={200}
  />
  <input
    className="thumb thumb--right"
    max={0}
    min={200}
    onChange={[Function]}
    type="range"
    value={0}
  />
  <div
    className="slider"
  >
    <div
      className="slider__track"
    />
    <div
      className="slider__range"
    />
    <div
      className="slider__left-value"
    >
      200
    </div>
    <div
      className="slider__right-value"
    >
      0
    </div>
  </div>
</div>
`;
